bin/movesrcfiles.js

```diff
@@ -0,0 +1,297 @@
+#! /usr/bin/env node
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+/**
+ * Moves the site files, and performs localization on a per release branch basis.
+ * How it works, for each of the release train branches:
+ *
+ *    1. it checks out the appropriate branch (e.g., nightly).
+ *
+ *    2. it sets up the appropriate directory structure:
+ *      ./build/version/branch/locale/
+ *
+ *    3. it copies over the files:
+ *      - the service worker.
+ *      - the js and css folders
+ *
+ *    4. Performs localization on HTML template (src/lang-tag/index.html)
+ *       using Handlebars.
+ *
+ *    5. Copies the localized HTML to the appropriate version/branch/locale
+ *       directory.
+ *
+ *  Best of all, files that don't need to be localized are just placed at the
+ *  "branch" level, so to avoid duplication. For example, there sw.js:
+ *   ./build/v2/nightly/sw.js
+ *
+ *  Localized HTML and the string.json files, then gets place in:
+ *    ./build/v2/nightly/ach/index.html
+ *    ./build/v2/aurora/ach/locale/strings.json
+ *
+ */
+/*jshint node:true, esnext: true*/
+"use strict";
+const async = require("marcosc-async");
+const path = require("path");
+const fetch = require("node-fetch"); // jshint ignore:line
+const fse = require("fs-extra");
+const handlebars = require("handlebars");
+const baseSrcDir = path.resolve(`${__dirname}/../src`);
+const outBaseDir = path.resolve(`${__dirname}/../build`);
+const l10nBaseDir = path.resolve(`${__dirname}/../l10n`);
+const exec = require("child_process").exec;
+const version = "v2";
+let workingBranch = "gh-pages";
+
+/**
+ * Basic wrapper for performing Git operations asynchronously.
+ *
+ * @type {Object}
+ */
+const Git = {
+  /**
+   * Checks out a git branch.
+   *
+   * @param  {String} branch The branch to checkout out.
+   * @return {Promise} Resolves when command is done executing.
+   */
+  checkout(branch) {
+    return this.promisedExec(`git checkout ${branch}`);
```

Comment:
Notwithstanding the errors when running the script, I think doing a git checkout from the script might not be what we want to do.

We want to use the branches for version numbers, not release branches just yet.
I know we have 2 dimensions of complexity: version number and release channel.

We have agreed to use branches for version numbers, however, we still need to figure out how to layer the channel dimension in the branching process.

Let's not worry about release channels just yet.